src/web/components/StickyBottomBanner/PuzzlesBanner.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useEffect, useState } from 'react';
+import * as emotion from 'emotion';
+import * as emotionCore from '@emotion/core';
+import * as emotionTheming from 'emotion-theming';
+import { useHasBeenSeen } from '@root/src/web/lib/useHasBeenSeen';
+import {
+	getWeeklyArticleHistory,
+	logView,
+} from '@root/node_modules/@guardian/automat-client';
+import {
+	shouldHideSupportMessaging,
+	getArticleCountConsent,
+} from '@root/src/web/lib/contributions';
+import {
+	sendOphanComponentEvent,
+	submitComponentEvent,
+} from '@root/src/web/browser/ophan/ophan';
+import { getCookie } from '@root/src/web/browser/cookie';
+import { getZIndex } from '@root/src/web/lib/getZIndex';
+import { trackNonClickInteraction } from '@root/src/web/browser/ga/ga';
+import { WeeklyArticleHistory } from '@root/node_modules/@guardian/automat-client/dist/types';
+import { CanShowResult } from '@root/src/web/lib/messagePicker';
+
+const checkForErrors = (response: Response) => {
+	if (!response.ok) {
+		throw Error(
+			response.statusText ||
+				`SlotBanner | An api call returned HTTP status ${response.status}`,
+		);
+	}
+	return response;
+};
+
+type HasBeenSeen = [boolean, (el: HTMLDivElement) => void];
+
+type BaseProps = {
+	isSignedIn: boolean;
+	contentType: string;
+	sectionName?: string;
+	shouldHideReaderRevenue: boolean;
+	isMinuteArticle: boolean;
+	isPaidContent: boolean;
+	isSensitive: boolean;
+	tags: TagType[];
+	contributionsServiceUrl: string;
+	alreadyVisitedCount: number;
+	engagementBannerLastClosedAt?: string;
+	subscriptionBannerLastClosedAt?: string;
+	weeklyArticleHistory?: WeeklyArticleHistory;
+};
+
+type BuildPayloadProps = BaseProps & {
+	countryCode: string;
+	hasConsentedToArticleCounts: boolean;
+};
+
+type CanShowProps = BaseProps & {
+	asyncCountryCode: Promise<string>;
+	remotePuzzlesBannerConfig: boolean;
+};
+
+// TODO specify return type (need to update client to provide this first)
+const buildPayload = (props: BuildPayloadProps) => {
```

Comment:
If this is only used in one place we might as well inline it? Saves a bit of scrolling around